packages/block-editor/src/components/block-title/index.js

```diff
@@ -40,5 +53,17 @@ export default function BlockTitle( { clientId } ) {
 		return null;
 	}
 
-	return blockType.title;
+	const { __experimentalLabel, title } = blockType;
+
+	// Check if it supports the label first. getBlockLabel will return the title
+	// if no label exists. We want to know if the label is undefined so that we
+	// can format it uniquely.
+	const label =
+		__experimentalLabel &&
+		truncate( getBlockLabel( blockType, attributes ), { length: 15 } );
+
+	if ( label ) {
+		return `${ title } - ${ label }`;
```

Comment:
Should we use the [colon](https://github.com/WordPress/gutenberg/pull/23847#issuecomment-658478033)? 🤔

Reply:
Oops! I guess that was a mindfart due to commiting this change late in the day 🙃